### ISSUE TITLE
Transliteration hook doesn't work with registrations ?

### ISSUE THREAD

**Issue by chrismfz** [hexonet/whmcs-ispapi-registrar#196]
For years we had a transliteration hook (in includes/hooks/) running with another provider.  We had a function to 
change from Greek to Greeklish beased on the Greeklish ISO ( https://en.wikipedia.org/wiki/ISO_843 )

https://docs.whmcs.com/Custom_Transliteration

Switching from them to hexonet, with the migration hook/plugin everything is fine. Transfers too.

But with ispapi for registrations and registrations only, this hook stopped / or the plugin ignores it ? I don't know.
Hexonet doesn't accept Greek characters and I am getting errors like:
Invalid attribute value; INVALID CONTACT [OWNERCONTACT (Invalid attribute value; Invalid char in NAME [Α])]
For contact name, city, address, and so on. 
The "A" it's just the first Greek character of the full name.

**Comment by chrismfz** [comment]
Hi papakai, we use this:

```
[root@pluto hooks]# cat greeklish.php
<?php

function hook_transliterate($string) {
    $string = str_replace("α","a",$string);
    $string = str_replace("ά","a",$string);
    $string = str_replace("Α","A",$string);
    $string = str_replace("Ά","A",$string);

    $string = str_replace("β","v",$string);
    $string = str_replace("Β","v",$string);

    $string = str_replace("α","a",$string);
    $string = str_replace("α","a",$string);

    $string = str_replace("Γ","G",$string);
    $string = str_replace("γ","g",$string);

    $string = str_replace("Δ","D",$string);
    $string = str_replace("δ","d",$string);

    $string = str_replace("ε","e",$string);
        $string = str_replace("Ε","E",$string);
        $string = str_replace("έ","e",$string);
        $string = str_replace("Έ","E",$string);

        $string = str_replace("Ζ","Z",$string);
        $string = str_replace("ζ","z",$string);

        $string = str_replace("Η","I",$string);
        $string = str_replace("η","i",$string);
        $string = str_replace("ή","i",$string);
        $string = str_replace("Ή","I",$string);

        $string = str_replace("Θ","Th",$string);
        $string = str_replace("θ","th",$string);

        $string = str_replace("Ι","I",$string);
        $string = str_replace("ι","i",$string);
        $string = str_replace("ί","i",$string);
        $string = str_replace("Ί","i",$string);

        $string = str_replace("Κ","K",$string);
        $string = str_replace("κ","k",$string);

        $string = str_replace("Λ","L",$string);
        $string = str_replace("λ","l",$string);

        $string = str_replace("Μ","M",$string);
        $string = str_replace("μ","m",$string);

        $string = str_replace("Ν","N",$string);
        $string = str_replace("ν","n",$string);

        $string = str_replace("Ξ","X",$string);
        $string = str_replace("ξ","x",$string);

        $string = str_replace("Ο","O",$string);
        $string = str_replace("ο","o",$string);
        $string = str_replace("Ό","O",$string);
        $string = str_replace("ό","o",$string);

        $string = str_replace("Π","P",$string);
        $string = str_replace("π","p",$string);

        $string = str_replace("Ρ","R",$string);
        $string = str_replace("ρ","r",$string);

        $string = str_replace("Σ","S",$string);
        $string = str_replace("σ","s",$string);
        $string = str_replace("ς","s",$string);

        $string = str_replace("Τ","T",$string);
        $string = str_replace("τ","t",$string);

        $string = str_replace("Υ","Y",$string);
        $string = str_replace("υ","y",$string);
        $string = str_replace("ύ","y",$string);
        $string = str_replace("Ύ","Y",$string);

        $string = str_replace("Φ","F",$string);
        $string = str_replace("φ","f",$string);

        $string = str_replace("Χ","Ch",$string);
        $string = str_replace("χ","ch",$string);

        $string = str_replace("Ψ","Ps",$string);
        $string = str_replace("ψ","ps",$string);

        $string = str_replace("Ω","O",$string);
        $string = str_replace("ω","o",$string);
        $string = str_replace("ώ","o",$string);
        $string = str_replace("Ώ","O",$string);

    return $string;
}

add_hook("Transliteration",1,"hook_transliterate");

?>
```

**Comment by papakai** [comment]
@chrismfz I've discussed this deeply with our backend engineers. Our systems are UTF-8 ready, but finally it depends on the registry provider behind that TLD. So, for gTLDs/nTLDs it might be working with language specific characters, but not with some ccTLDs.

I'll dive into reviewing our module to either considering $params (the transliterated data) or to provide a build-in possibility (how ever it might look like). I'll keep you posted.

**Comment by papakai** [comment]
@chrismfz this is in work and results are already looking good. We will start shipping the transliteration with our registrar module.
In addition to the greeklish transliteration it will also cover replacing html entities with their utf-8 representations.

It will just need a bit of time until we release. The next update is coming with huge improvements regarding additional domain fields and updating contact information.

**Comment by chrismfz** [comment]
Only the greeklish! 
Thank you <3 !

**Comment by papakai** [comment]
btw., having `add_hook("Transliteration",1,"hook_transliterate");` in place isn't required any longer. That's what they point out in the docs. I tested it, and yes it really works without that line of code.

**Comment by papakai** [comment]
Released as v6.0.4. Read my Notes [here](https://github.com/hexonet/whmcs-ispapi-registrar/releases/tag/v6.0.4) before upgrading.
